File: src/SearchService/Entities/Auction.cs
﻿using MongoDB.Entities;

namespace SearchService.Entities
{
    public class Auction : Entity
    {
        public int ReservePrice { get; set; }
        public string Seller { get; set; }
        public string Winner { get; set; }
        public int SoldAmount { get; set; }
        public int CurrentHighBid { get; set; }
        public string Status { get; set; }
        public DateTime DateStart { get; set; }
        public DateTime DateEnd { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string ImageUrl { get; set; }
        public string Category { get; set; }
        public DateTime DateCreated { get; set; }
        public DateTime DateUpdated { get; set; }
    }
}

File: frontend/app/types/Auction.ts
export type Auction = {
	reservePrice: number;
	seller: string;
	winner?: string;
	soldAmount?: number;
	currentHighBid: number;
	status: string;
	dateStart: string;
	dateEnd: string;
	title: string;
	description: string;
	imageUrl: string;
	category: string;
	dateCreated: string;
	dateUpdated: string;
	id: string;
};
